app/uk/gov/hmrc/securemessage/controllers/SecureMessageController.scala

```diff
@@ -309,13 +311,27 @@ class SecureMessageController @Inject() (
           Future.successful(InternalServerError)
         case Right(Some(message)) =>
           logger.debug(s"Secure Message is Read $id")
-          auditMessageReadStatus(message)
-          Future.successful(Ok)
+          val audit = auditMessageReadStatus(message).recover { case e =>
+            logger.error("Audit for setReadTime is failed ")
+          }
+          val stats = updateStats(message).recover { case e => logger.error("Stats update for setReadTime is failed") }
+
+          Future.sequence(Seq(audit, stats)).map(_ => Ok)
+
         case Right(None) => Future.successful(InternalServerError(s"failed to set read time for: $id"))
       }
       .recoverWith { case NonFatal(e) =>
         logger.error(s"Failed to set read time for: $id due to ${e.getMessage}")
         Future.successful(InternalServerError(s"Failed to set read time for: $id due to ${e.getMessage}"))
       }
   }
+
+  private def updateStats(message: Message): Future[Unit] = {
+    val (identifier, formId) = message match {
+      case s: SecureMessage => (s.recipient.identifier.name, s.details.map(_.formId).getOrElse("NoForm"))
+      case l: Letter        => (l.recipient.identifier.name, l.body.flatMap(_.form).getOrElse("NoForm"))
+      case _                => throw IllegalAccessException(s"Invalid message: ${message.getClass}")
```

Comment:
Do we need to throw exception when we try to log stats? may be log like an error should do?

Reply:
@muralishub It is not just to log the error, basically it returns an error response to the caller. Though it is an edge case, we want to return something if it happens. Hope it helps.

Reply:
I think I read your comment wrong earlier. Yes, you're right. We dont require to catch the exception here, we can suppress it by log. Thanks for pointing itoput. Made the update, please check it again. Thanks